src/Faker/ORM/Propel2/ColumnTypeGuesser.php

```diff
@@ -50,29 +50,26 @@ public function guessFormat(ColumnMap $column)
                     return $generator->randomNumber($size + 2) / 100;
                 };
             case PropelTypes::TINYINT:
-                return function () {
-                    return mt_rand(0, 127);
+                return function () use ($generator) {
+                    return $generator->numberBetween(0, 127);
                 };
             case PropelTypes::SMALLINT:
-                return function () {
-                    return mt_rand(0, 32767);
+                return function () use ($generator) {
+                    return $generator->numberBetween(0, 32767);
                 };
             case PropelTypes::INTEGER:
-                return function () {
-                    return mt_rand(0, intval('2147483647'));
+                return function () use ($generator) {
+                    return $generator->numberBetween(0, 2147483647);
                 };
             case PropelTypes::BIGINT:
-                return function () {
-                    return mt_rand(0, intval('9223372036854775807'));
+                return function () use ($generator) {
+                    return $generator->numberBetween(0, 9223372036854775807);
```

Comment:
This should be `0, INT_MAX`, surely? Otherwise this code is broken on 32-bit builds of PHP.

Reply:
(because that int gets cast to a float, then the wrong type is passed to numberBetween.

Reply:
This should indeed be `INT_MAX`.

Reply:
@GrahamCampbell @bramceulemans Thank you for review, changes applied.